website/src/wordpress-page.tsx

```diff
@@ -41,6 +46,12 @@ export const WordpressContents = ({ content }: { content: string }) => {
 
 const parseOptions: HTMLReactParserOptions = {
   replace(node) {
+    if (isText(node) && node.data.startsWith("[") && node.data.endsWith("]") && node.data.indexOf(":") !== -1) {
+      const text = node.data.split(/[\[,\],:,-]/).filter((x) => x !== "") // figure out why empty strings in start and end of array
```

Comment:
How about instead of using `split`, you use capture groups in the regex? That should be more robust and follow a more explicit syntax. With your current code, this footnote-like text would pass your checks: `[that book]: found at the library--with dust on top` and turn into `["that book", "found at the library", "", "with dust on top"]`. Which means it's not strict enough. What you're reaching for is predefining a regex pattern for the whole reference, then checking if the node text matches the pattern, then if so using the matching results. [Here's a doc on captures in JS regex](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Guide/Regular_Expressions/Groups_and_Ranges)